### DISCUSSION TITLE
Cross referencing arbitrary location

### DISCUSSION THREAD

**Discussion by madjxatw** [mkdocs/mkdocs#3566] [Q&A] [answered]
Hi, does mkdocs support cross referencing arbitrary locations like Sphinx in current page or another page? If not, is there any plugin that offers this feature?

**Comment by pawamoy** [comment, marked as answer]
Yes, check out [mkdocs-autorefs](https://mkdocstrings.github.io/autorefs/) (and [mkdocstrings](https://mkdocstrings.github.io/usage/#cross-references) for the API side, which uses autorefs internally).

**Reply by madjxatw** [reply]
autorefs seems to support referencing headings only, but not arbitrary location.

**Reply by pawamoy** [reply]
There's a PR in autorefs that would enable that, by adding an option to scan anchors in the page HTML. You could then add anchors with `<a id="my-anchor"></a>` (or in Markdown `[](){ #my-anchor }`) anywhere in a file, and cross-ref them with `[my-anchor][]` or `[My anchor][my-anchor]`. Let us know if you'd be interested in such a feature.

**Reply by madjxatw** [reply]
So this PR hasn't be merged yet, right? I really need such a feature for quick jump to a specific location in a section (maybe far away from the heading starting the section) in another page, thanks!

**Reply by pawamoy** [reply]
It's not merged and it's not even ready to merge, but we can try to find some time to finish it in the coming days/weeks. In the meantime feel free to check out other plugins in the [catalog](https://github.com/mkdocs/catalog), particularly in the "Links & references" category: maybe one of them already supports that.

**Reply by madjxatw** [reply]
Thanks in advance! I would mark your reply as answer once I confirm it works.

**Reply by madjxatw** [reply]
I've given a try to mkdocs-autorefs 1.0.1 and it works a like a charm, thanks!

**Reply by pawamoy** [reply]
Amazing, thanks for letting us know!
